shell/platform/android/test/io/flutter/view/AccessibilityBridgeTest.java

```diff
@@ -0,0 +1,188 @@
+    AccessibilityBridge accessibilityBridge = setUpBridge();
+
+    TestSemanticsNode testSemanticsNode = new TestSemanticsNode();
+    testSemanticsNode.label = "Hello, World";
+    TestSemanticsUpdate testSemanticsUpdate = testSemanticsNode.toUpdate();
+
+    accessibilityBridge.updateSemantics(testSemanticsUpdate.buffer, testSemanticsUpdate.strings);
+    AccessibilityNodeInfo nodeInfo = accessibilityBridge.createAccessibilityNodeInfo(0);
+
+    assertEquals(nodeInfo.getContentDescription(), "Hello, World");
+    assertEquals(nodeInfo.getText(), null);
+  }
+
+  @Test
+  public void itDescribesTextFieldsWithText() {
+    AccessibilityBridge accessibilityBridge = setUpBridge();
+
+    TestSemanticsNode testSemanticsNode = new TestSemanticsNode();
+    testSemanticsNode.label = "Hello, World";
+    testSemanticsNode.addFlag(AccessibilityBridge.Flag.IS_TEXT_FIELD);
+    TestSemanticsUpdate testSemanticsUpdate = testSemanticsNode.toUpdate();
+
+    accessibilityBridge.updateSemantics(testSemanticsUpdate.buffer, testSemanticsUpdate.strings);
+    AccessibilityNodeInfo nodeInfo = accessibilityBridge.createAccessibilityNodeInfo(0);
+
+    assertEquals(nodeInfo.getContentDescription(), null);
+    assertEquals(nodeInfo.getText(), "Hello, World");
+  }
+
+  @Test
+  public void itDoesNotContainADescriptionIfScopesRoute() {
+    AccessibilityBridge accessibilityBridge = setUpBridge();
+
+    TestSemanticsNode testSemanticsNode = new TestSemanticsNode();
+    testSemanticsNode.label = "Hello, World";
+    testSemanticsNode.addFlag(AccessibilityBridge.Flag.SCOPES_ROUTE);
+    TestSemanticsUpdate testSemanticsUpdate = testSemanticsNode.toUpdate();
+
+    accessibilityBridge.updateSemantics(testSemanticsUpdate.buffer, testSemanticsUpdate.strings);
+    AccessibilityNodeInfo nodeInfo = accessibilityBridge.createAccessibilityNodeInfo(0);
+
+    assertEquals(nodeInfo.getContentDescription(), null);
+    assertEquals(nodeInfo.getText(), null);
+  }
+
+  AccessibilityBridge setUpBridge() {
+    View view = mock(View.class);
+    Context context = mock(Context.class);
+    when(view.getContext()).thenReturn(context);
+    when(context.getPackageName()).thenReturn("test");
+    AccessibilityChannel accessibilityChannel = mock(AccessibilityChannel.class);
+    AccessibilityManager accessibilityManager = mock(AccessibilityManager.class);
+    ContentResolver contentResolver = mock(ContentResolver.class);
+    PlatformViewsAccessibilityDelegate platformViewsAccessibilityDelegate =
+        mock(PlatformViewsAccessibilityDelegate.class);
+    AccessibilityBridge accessibilityBridge =
+        new AccessibilityBridge(
+            view,
+            accessibilityChannel,
+            accessibilityManager,
+            contentResolver,
+            platformViewsAccessibilityDelegate);
+    return accessibilityBridge;
+  }
+
+  /// The encoding for semantics is described in platform_view_android.cc
+  class TestSemanticsUpdate {
```

Comment:
if we want this mapping to be maintained manually, can we add notes to each of accessibilitybridge.updatesemantics, platformviewandroid.updatesemantics and here to remind maintainers?

Reply:
Something like : "This logic is also used in the test fixture at path/to/test.java"?

Reply:
Sure. Something like "if you change stuff here, don't forget to update the test file as well"

Reply:
I added a comment to platform_view_android.cc, since that defines the encoding format. Elsewhere it is only consumed